lib/yardstick/ordered_set.rb

```diff
@@ -17,7 +17,7 @@ class OrderedSet
     def initialize(entries = nil)
       @entries = []
       @index   = {}
-      merge(entries || [])
+      merge(entries || self)
```

Comment:
I opted for `initialize(entries = nil)` over the `Undefined` option because the `Undefined` option left me with the mutation

```diff
- def initialize(entries = Undefined)
+ def initialize(entries = self)
    @entries = []
    @index   = {}
    merge(entries) unless entries.equal?(Undefined)
  end
```

which would be difficult to kill.  Also, you noted 

> Although one thing this doesn't protect against is cases where the caller passes in nil explicitly. 

but in this case `Yardstick::OrderedSet.new(nil)` would raise an exception anyways since `merge` expects `entries` to respond to `each`